### PR TITLE
[core] Introduce RenderTree and RenderOrchestrator

### PR DESCRIPTION
**Pull request by pozdnyakov**
- `RenderOrchestrator` is a class responsible for preparing map render data (`RenderTree`), orchestration tasks and implementation of the queryRenderedFeatures API. Orchestrator also notifies map of the resources updates and errors 
- `RenderTree` class represents a snapshot of the data, required for rendering a map at a moment of time. The map rendering code shall operate with `RenderTree` instances.

Part of the implementation for https://github.com/mapbox/mapbox-gl-native/issues/14638

### CODE REVIEW

**Comment by pozdnyakov** [comment]
Please take a look

File: src/mbgl/renderer/renderer_impl.cpp [hunk 1]
```diff
@@ -454,12 +148,12 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
         const auto debugGroup(parameters.renderPass->createDebugGroup("opaque"));
 
         uint32_t i = 0;
-        for (auto it = renderItems.rbegin(); it != renderItems.rend(); ++it, ++i) {
+        for (auto it = layerRenderItems.rbegin(); it != layerRenderItems.rend(); ++it, ++i) {
```
Comment (tmpsantos) [inline src/mbgl/renderer/renderer_impl.cpp]: This sounds like a good use case for a `std::for_each` with a functor that would keep the state for `i`.

```
struct OpaquePass {
  void operator()(RenderItem&) { ...}
  int currentLayer{0};
};
```

File: src/mbgl/renderer/render_tree.hpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+namespace mbgl {
+
+class PaintParameters;
+class PatternAtlas;
+
+namespace gfx {
+class UploadPass;
+} // namespace gfx
+
+class RenderItem {
```
[the remark this answers is not in the document]
Comment (pozdnyakov) [reply src/mbgl/renderer/render_tree.hpp]: Renderer operates with both layers and sources via this interface |=> it guaranties renderer can only do upload and renderering.
Further, we could introduce `RenderPass::Debug` and merge layers and sources render items, but need to profile performance drawback before we embark on that.

File: src/mbgl/renderer/renderer_impl.cpp (hunk 1 above)
Comment (pozdnyakov) [reply src/mbgl/renderer/renderer_impl.cpp]: wouldn't it be an overkill (given that we've several passes |=> would need several functors) ? :thinking:

**Pull request merged**